src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_START_CAPASITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private float threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_START_CAPASITY];
+        threshold = table.length * DEFAULT_LOAD_FACTOR;
+    }
+
+    private int hash(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+
+        }
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = hash(newNode.key);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> oldNode = table[index];
+        while (oldNode != null) {
+            if (Objects.equals(oldNode.key, key)) {
+                oldNode.value = value;
+                return;
+            }
+            if (oldNode.next == null) {
+                oldNode.next = newNode;
+                size++;
+                return;
+            }
+            oldNode = oldNode.next;
+        }
+    }
 
+    private void resize() {
+        threshold *= 2;
+        Node<K, V>[] oldTable = table;
+        table = new Node[oldTable.length * 2];
```

Comment:
let;s use here a constant variable and also move this private helper under all public
```suggestion
        threshold *= RESIZE_FACTOR;
        Node<K, V>[] oldTable = table;
        table = new Node[oldTable.length * RESIZE_FACTOR];
```